proposals/0001-unify-nat-and-natural.md

```diff
@@ -0,0 +1,71 @@
+---
+author: Rinat Stryungis
+date-accepted: ""
+ticket-url: "https://gitlab.haskell.org/ghc/ghc/-/issues/10776"
+implemented: "Yes"
+---
+
+This proposal is [discussed at this pull request](https://github.com/haskell-core/core-libraries-proposals/pull/0>).
+**After creating the pull request, edit this file again, update the number in
+the link, and delete this bold sentence.**
+
+# Proposal title
+
+The changes in the proposal remove the separate built-in kind `Nat` 
+in favor of promoted type `Natural`.
+
+
+## Motivation
+
+Existence of separated kind `Nat` and impossibility of promotion 
+of usual `Natural` data type (and data constructors with `Natual` fields)
+looks like an unnecessary and inconvenient complication. 
+It forces to create redundant types with `Nat` fields in their constructors (it makes these types uninhabited in terms)
+only for promotional purposes. This is more cumbersome and less safe than the promotion 
```

Comment:
```suggestion
only for the sake of promotion. This is more cumbersome and less safe than the promotion 
```

Reply:
By the way, how is it less safe? I think it's just more cumbersome.

Reply:
It is less safe because now one has to manually create 2 synchronized types and you could make a mistake or forget to change one of them in case of refactoring for example.

Reply:
There's a technique to avoid this, you would typically write something like this:

```
data MyType' n = MkT X Y n Z

type MyType = MyType' Natural  -- for use in terms
type MyTypeP = MyType' Nat     -- for use in types
```

So you're not really repeating the `X Y Z` part of the declaration.

Reply:
Yea, but it is inconvenient.... ok you are right :-)